modules/ml/include/opencv2/ml.hpp

```diff
@@ -1838,6 +1867,166 @@ CV_EXPORTS void randMVNormal( InputArray mean, InputArray cov, int nsamples, Out
 CV_EXPORTS void createConcentricSpheresTestSet( int nsamples, int nfeatures, int nclasses,
                                                 OutputArray samples, OutputArray responses);
 
+/** @brief Artificial Neural Networks - Multi-Layer Perceptrons.
+
+@sa @ref ml_intro_ann
+*/
+class CV_EXPORTS_W ANN_MLP_ANNEAL : public ANN_MLP
+{
+public:
+    CV_WRAP virtual void setTrainMethod(int method, double param1 = 0, double param2 = 0, double param3 = 0, int param4 = 0) = 0;    /** ANNEAL: Update initial temperature.
+    It must be \>=0. Default value is 10.*/
+    /** @see setAnnealInitialT */
+    CV_WRAP virtual double getAnnealInitialT() const=0;
```

Comment:
Because you should not use `=0` here.
You should provide implementations for these methods in .cpp file.

Reply:
![image](https://user-images.githubusercontent.com/3591626/33834628-d69c6f90-de83-11e7-92c0-f4b230cbfd62.png)